src/System.Linq.Queryable/src/System/Linq/Queryable.cs

```diff
@@ -39,10 +39,10 @@ public static IQueryable<TSource> Where<TSource>(this IQueryable<TSource> source
             return source.Provider.CreateQuery<TSource>(
                 Expression.Call(
                     null,
-                    GetMethodInfoOf(() => Queryable.Where(
+                    GetMethodInfoOf(() => Where(
```

Comment:
At the very least, it seems a strange approach compared to going through a `Func` which could be created by having GetMethodInfoOf be like:

```C#
private static MethodInfo GetMethodInfo<T1, T2>(Func<T1, T2> f, T1 _0) {
            return f.Method;
}
```

And calling `GetMethodOf(Where, predicate)`, with of course overloads for other arities. I would have imagined that would be cheaper.

Caching could either cache just the generic method definition, or each instantiated method. The former would still have to call `MakeGenericMethod()` while the latter could grown large in the face of dynamic use.